src/main/java/io/cdap/plugin/cdc/source/sqlserver/ResultSetToDMLRecord.java

```diff
@@ -40,57 +40,120 @@
     Map<String, Object> changes = new HashMap<>();
     for (int i = 0; i < changeSchema.getFields().size(); i++) {
-      int column = i + CHANGE_TABLE_COLUMNS_SIZE;
+
+      Schema.Field field = changeSchema.getFields().get(i);
+      int column = getColumnForFeild(metadata, field.getName());
       int sqlType = metadata.getColumnType(column);
+      String sqlTypeName = metadata.getColumnTypeName(column);
       int sqlPrecision = metadata.getPrecision(column);
       int sqlScale = metadata.getScale(column);
-      Schema.Field field = changeSchema.getFields().get(i);
-      Object sqlValue = DBUtils.transformValue(sqlType, sqlPrecision, sqlScale, resultSet, field.getName());
-      Object javaValue = transformSQLToJavaType(sqlValue);
-      changes.put(field.getName(), javaValue);
+      getColumnForFeild(metadata, field.getName());
+
+      try {
+        /**
+         * Handling clob and blob data type ... JTDS does not support free() that throws exception from the DBUtils.
+         */
+        if (sqlType == 2005) {
+          Clob clob = resultSet.getClob(field.getName());
+          String retVal = (clob != null ? clob.getSubString(1, (int) clob.length()) : null);
+          changes.put(field.getName(), "\"" + retVal.toString()  + "\"");
+        } else if (sqlType == 2004) {
+          Blob blob = resultSet.getBlob(field.getName());
+          byte[] blobVal = (blob != null ? blob.getBytes(1L, (int) blob.length()) : null);
+          changes.put(field.getName(), "\"" + blobVal.toString()  + "\"");
+        } else if (sqlType == -8) {
+          Object sqlValue = DBUtils.transformValue(sqlType, sqlPrecision, sqlScale, resultSet, field.getName());
+          Object javaValue = transformSQLToJavaType(sqlValue);
+          changes.put(field.getName(), "\"" + javaValue.toString() + "\"");
+        } else {
+          Object sqlValue = DBUtils.transformValue(sqlType, sqlPrecision, sqlScale, resultSet, field.getName());
+          Object javaValue = transformSQLToJavaType(sqlValue);
+          changes.put(field.getName(), "\"" + javaValue.toString() + "\"");
+        }
+      } catch (Exception e) {
+        if (resultSet != null && resultSet.getObject(column) != null) {
+          changes.put(field.getName(), "\"" + resultSet.getObject(column).toString() + "\"");
+        } else {
+          changes.put(field.getName(), null);
+        }
+      }
     }
     return changes;
   }
 
-  private static Schema getChangeSchema(ResultSet resultSet) throws SQLException {
+  private static int getColumnForFeild(ResultSetMetaData metadata, String columnName) throws Exception {
+    for (int i = 1; i <= metadata.getColumnCount(); i++) {
+      if (metadata.getColumnLabel(i).equals(columnName) || metadata.getColumnName(i).equals(columnName)) {
+        return i;
+      }
+    }
+    throw new Exception ("Can not find " + columnName);
+
+  }
+  private static Schema getChangeSchema(ResultSet resultSet, int size) throws Exception {
```

Comment:
as mentioned earlier, this is an existing bug but Schema objects should be re-used when possible. Otherwise this will generate a bunch of extra memory usage.

Reply:
agreed